CMake/RealmCore.cmake

```diff
@@ -40,7 +40,7 @@ endif()
 
 function(use_realm_core version_or_path_to_source)
     if("${version_or_path_to_source}" MATCHES "^[0-9]+(\\.[0-9])+")
-        if(APPLE OR REALM_PLATFORM STREQUAL "Android")
+        if(APPLE OR LINUX OR REALM_PLATFORM STREQUAL "Android")
```

Comment:
I think a lot of the `LINUX` checks should really be either checking CMake's existing `UNIX` variable or something more specific, such as detecting whether epoll is supported.